library/alloc/src/boxed.rs

```diff
@@ -1361,6 +1367,27 @@ impl From<&str> for Box<str> {
 #[cfg(not(no_global_oom_handling))]
 #[stable(feature = "box_from_cow", since = "1.45.0")]
 impl From<Cow<'_, str>> for Box<str> {
+    /// Converts a `Cow<'_, str>` into a `Box<str>`
+    ///
+    /// This conversion allocates on the heap
+    /// and performs a copy of `cow`.
```

Comment:
Same here. It doesn't necesarily allocate or copy anything for a `Cow::Owned`.

Reply:
Fixed in e41d5ae65c0f51ca459064808961e6d59de6f6c4